src/basic/process-util.c

```diff
@@ -1732,6 +1733,48 @@ int make_reaper_process(bool b) {
         return 0;
 }
 
+int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp, pid_t *ret_pid) {
+        posix_spawnattr_t attr;
+        sigset_t mask;
+        pid_t pid;
+        int r;
+
+        /* Forks and invokes 'path' with 'argv' and 'envp' using CLONE_VM and CLONE_VFORK, which means the
+         * caller will be blocked until the child either exits or exec's. The memory of the child will be
+         * fully shared with the memory of the parent, so that there are no copy-on-write or memory.max
+         * issues. */
+
+        assert(path);
+        assert(argv);
+        assert(ret_pid);
+
+        assert_se(sigemptyset(&mask) >= 0);
+
+        r = posix_spawnattr_init(&attr);
+        if (r != 0)
+                return -r; /* These functions return a positive errno on failure */
+        r = posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGMASK);
```

Comment:
btw, you might as well set POSIX_SPAWN_SETSIGDEF here too, if you touch signals. i.e. set the signal handler funcs to defaults too.

Reply:
done